src/python/gem5/simulate/simulator.py

```diff
@@ -282,12 +283,79 @@ def set_max_ticks(self, max_tick: int) -> None:
             raise ValueError(
                 f"Max ticks must be less than {m5.MaxTick}, not {max_tick}"
             )
+        if self._hypercall_max_ticks:
+            warn(
+                "A hypercall 6 exit has already been scheduled for tick "
+                f"{self._hypercall_max_ticks}. Setting hypercall and classic "
+                "exits in the same simulation is not well tested and the "
+                "simulation may not behave as expected."
+            )
+
         self._max_ticks = max_tick
 
     def get_max_ticks(self) -> int:
         assert hasattr(self, "_max_ticks"), "Max ticks not set"
         return self._max_ticks
 
+    def set_hypercall_absolute_max_ticks(
+        self, max_tick: int, exit_str: str = "Max ticks reached"
+    ) -> None:
+        """Set the maximum number of ticks to simulate before the simulation
+        exits with a hypercall 6 exit. This exit will be handled by
+        ScheduledExitEventHandler by default. See `src/python/gem5/simulate/
+        exit_handler.py` for details.
+
+        Args:
+            max_tick (int): The number of ticks to simulate before exiting.
+            exit_str (str): The reason for the exit.
+        """
+        if self._max_ticks and self._max_ticks != m5.MaxTick:
+            warn(
+                "A classic MAX_TICKS exit has already been scheduled for "
+                f"tick {self._max_ticks}. Setting hypercall and classic "
+                "exits in the same simulation is not well tested and the "
+                "simulation may not behave as expected."
+            )
+        if exit_str == "Tick exit reached":
+            warn(
+                f"The exit string '{exit_str}' will cause the simulation "
+                "to exit with a classic ExitEvent.SCHEDULED_TICK exit event "
+                "instead of a hypercall."
+            )
+        m5.scheduleTickExitAbsolute(max_tick, exit_str)
+        self._hypercall_max_ticks = max_tick
+
+    def set_hypercall_relative_max_ticks(
```

Comment:
Sorry one other thing... why don't you just call `set_hypercall_absolute_max_ticks(curTick() + ticks_from_current)` and remove the duplicated warnings.